common/src/main/java/com/wynntils/features/user/overlays/PowderAbilityBarOverlayFeature.java

```diff
@@ -0,0 +1,134 @@
+import com.wynntils.core.features.overlays.sizes.GuiScaledOverlaySize;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.objects.CustomColor;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.item.parsers.WynnItemMatchers;
+import com.wynntils.wynn.model.ActionBarModel;
+import com.wynntils.wynn.objects.Powder;
+
+public class PowderAbilityBarOverlayFeature extends UserFeature {
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay powderAbilityBarOverlay = new PowderAbilityBarOverlay();
+
+    @Override
+    protected void onInit(
+            ImmutableList.Builder<Condition> conditions, ImmutableList.Builder<Class<? extends Model>> dependencies) {
+        dependencies.add(ActionBarModel.class);
+    }
+
+    public static class PowderAbilityBarOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        @Config
+        public boolean flip = false;
+
+        @Config
+        public boolean onlyIfWeaponHeld = true;
+
+        @Config
+        public boolean hideIfNoCharge = false;
+
+        public PowderAbilityBarOverlay() {
+            super(
+                    new OverlayPosition(
+                            -30,
+                            150,
+                            VerticalAlignment.Bottom,
+                            HorizontalAlignment.Center,
+                            OverlayPosition.AnchorSection.BottomMiddle),
+                    new GuiScaledOverlaySize(81, 21));
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            float powderSpecialCharge = ActionBarModel.getPowderSpecialCharge();
+            Powder powderSpecialType = ActionBarModel.getPowderSpecialType();
+            if (this.onlyIfWeaponHeld
+                    && !WynnItemMatchers.isWeapon(McUtils.inventory().getSelected())) return;
+            if (this.hideIfNoCharge && powderSpecialCharge == 0) return;
+
+            renderWithSpecificSpecial(poseStack, powderSpecialCharge, powderSpecialType);
+        }
+
+        @Override
+        public void renderPreview(PoseStack poseStack, float partialTicks, Window window) {
+            renderWithSpecificSpecial(poseStack, 40, Powder.THUNDER);
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {}
+
+        private void renderWithSpecificSpecial(
+                PoseStack poseStack, float powderSpecialCharge, Powder powderSpecialType) {
+            Texture universalBarTexture = Texture.UNIVERSAL_BAR;
```

Comment:
It's a bit weird that this overlay does not use the same types as the health and mana bar, but that is likely acceptable.